lib/src/Uncertainty/Distribution/GeneralizedExtremeValueFactory.cxx

```diff
@@ -82,6 +94,899 @@ GeneralizedExtremeValue GeneralizedExtremeValueFactory::buildAsGeneralizedExtrem
+  UnsignedInteger nP = 0;
+  Point x0;
+  for (UnsignedInteger i = 0; i < 3; ++ i)
+  {
+    const UnsignedInteger nI = basisCollection[i].getSize();
+    nP += nI;
+
+    // initialize first coefficient of basis to 1, 0 elsewhere
+    Point x0i(nI);
+    x0i[0] = 1.0;
+    x0.add(x0i);
+
+    const Description betaVars(Description::BuildDefault(nI, "beta"));
+    const Description fVars(Description::BuildDefault(nI, "f"));
+    Description inputVars(betaVars);
+    inputVars.add(fVars);
+    String formula;
+    for (UnsignedInteger j = 0; j < nI; ++ j)
+    {
+      formula += betaVars[j] + " * " + fVars[j];
+      if (j < nI - 1)
+        formula += " + ";
+    }
+    const SymbolicFunction linearCombination(inputVars, Description(1, formula));
+    Indices betaIndices(nI);
+    betaIndices.fill();
+    const ParametricFunction parametric(linearCombination, betaIndices, Point(nI));
+    Collection<Function> coll(nI);
+    for (UnsignedInteger j = 0; j < nI; ++ j)
+      coll[j] = basisCollection[i][j];
+    const AggregatedFunction aggregated(coll);
+    ComposedFunction composed(parametric, aggregated);
+    composed.setOutputDescription({build().getParameterDescription()[i] + "(t)"});
+    thetaFunctions[i] = composed;
+  }
+  Function thetaFunction = AggregatedFunction(thetaFunctions);
+  if (inverseLinkFunction.getEvaluation().getImplementation()->isActualImplementation())
+    thetaFunction = ComposedFunction(inverseLinkFunction, thetaFunction);
+
+  const GeneralizedExtremeValueTimeVaryingLikelihoodEvaluation evaluation(sample, grid, thetaFunction, basisCollection);
+  const Function objective(evaluation.clone());
+  OptimizationProblem problem(objective);
+  problem.setMinimization(false);
+
+  // heuristic for feasible mu
+  UnsignedInteger i = 0;
+  const UnsignedInteger maxIter = ResourceMap::GetAsUnsignedInteger("GeneralizedExtremeValueFactory-FeasibilityMaximumIterationNumber");
+  const Scalar rho = ResourceMap::GetAsScalar("GeneralizedExtremeValueFactory-FeasibilityRhoFactor");
+  while ((evaluation(x0)[0] == -SpecFunc::MaxScalar) && (i < maxIter))
+  {
+    x0[0] *= rho;
+    ++ i;
+  }
+
+  TNC solver(problem);
+  solver.setProblem(problem);
+  solver.setMaximumEvaluationNumber(ResourceMap::GetAsUnsignedInteger("GeneralizedExtremeValueFactory-MaximumEvaluationNumber"));
+  solver.setStartingPoint(x0);
+  solver.run();
+  const Point optimalParameter(evaluation.uToX(solver.getResult().getOptimalPoint()));
+
+  // estimate parameter distribution via the Fisher information matrix
+  const UnsignedInteger size = sample.getSize();
+  Matrix fisher(nP, nP);
+  const Scalar epsilon = ResourceMap::GetAsScalar("Evaluation-ParameterEpsilon");
+  for (UnsignedInteger i = 0; i < size; ++ i)
+  {
+    thetaFunction.setParameter(optimalParameter);
+    const Scalar pdfIRef = buildAsGeneralizedExtremeValue(thetaFunction(grid[i])).computePDF(sample[i]);
+
+    // evaluate dpdf/dbeta by finite-differences
```

Comment:
No other option than finite differences here?

Reply:
we would need to rewrite it as a function, the blocking part it the GEV PDF gradient, 
I dont want to take the risk to rewrite GeneralizedExtremeValue::computePDFGradient